README.rst

```diff
@@ -100,15 +100,27 @@ file within that directory contains a refresh token for the user who authorized
 themselves. This refresh token should be treated as if it were a password and
 not shared or otherwise disclosed!
 
-To find the course id, item id, and part id, first go to the web authoring
-interface for your programming assignment. There, the URL will be of the form:
-``/:courseSlug/author/outline/programming/:itemId/``. The part id will be
-displayed in the authoring user interface for each part. To convert the
-``courseSlug`` into a ``courseId``, you can take advantage of our `Course API` putting in the appropriate ``courseSlug``. For example, given a
-course slug of ``developer-iot``, you can query the course id by making the
-request: ``https://api.coursera.org/api/onDemandCourses.v1?q=slug&slug=developer-iot``.
-The response will be a JSON object containing an ``id`` field with the value:
-``iRl53_BWEeW4_wr--Yv6Aw``.
+To find the course/branch id, item id, and part id:
+
+1. Go to the web authoring interface for your programming assignment. There, the URL will be of the form::
+  
+    /:courseSlug/author/outline/programming/:itemId/
+    
+The ``part id`` can be found in the authoring user interface for each part. 
+
+2. To convert the ``courseSlug`` into a ``courseId``, you can take advantage of our `Course API` putting in the appropriate ``courseSlug``. For example, given a course slug of ``developer-iot``, you can query the course id by making the request::
+
+    https://api.coursera.org/api/onDemandCourses.v1?q=slug&slug=developer-iot
+    
+The response will be a JSON object containing an ``id`` field with the value::
+
+    iRl53_BWEeW4_wr--Yv6Aw
+    
+3. If the grader that you're looking to upload/update is not in the `Original Version` of the course, you'll most likely need to get a branch id. To this, you'll need the ``courseId`` and the ``versionName``. Make the following request::
+
+    https://api.coursera.org/api/authoringBranchProperties.v1?q=course&courseId={course_id}&fields=properties
+    
+Search for your version name (it'll be in the ``properties.name`` field), and find the associated ``id`` (it should look something like: ``authoringBranch~xxxxxxxxxxxxxxxxxxxxxx``.
```

Comment:
Will it be always `authoringBranch~xxx` for base versions as well?

Reply:
I believe so. That's what I heard from the person I spoke with at Coursera, and the data I've seen bears this out.

Reply:
The original version won't have the `authoringBranch` prefix, but all others will

Reply:
OK, in that case, it might be helpful to provide that distinction here.